### PR TITLE
DHFPROD-4947:Provide info when main.sjs is not found or user doesn't required permissions

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Provide info when main.sjs is not found or user doesn't required permissions
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs [hunk 1]
```diff
@@ -116,7 +116,7 @@ class Step {
     try {
       stepModule = this.hubUtils.retrieveModuleLibrary(moduleUri);
     } catch (e) {
-      httpUtils.throwBadRequest(`Unable to access module: ${moduleUri}. Verify that this module is in your modules database and that your user account has a role that grants read permission to this module.`);
+      httpUtils.throwBadRequest(`Unable to access module: ${moduleUri}. Verify that this module is in your modules database and that your user account has a role that grants read and execute permission to this module`);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs:119]: @srinathgit  Can you test one more thing - see what happens if the step module is present but has a compilation error in it. I think we may get the same error, in which case we really ought to include "e.stack" at the end of the message - e.g. "; cause: ${e.stack}".

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs (hunk 1 above)
Comment (srinathgit) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs:119]: We cannot have the same message if there is compilation error right ? I have changed the message to 
```
`Cannot run module: ${moduleUri}; cause: ${e.stack}`
```
Let me know if that's ok.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs [hunk 2]
```diff
@@ -116,7 +116,10 @@ class Step {
     try {
       stepModule = this.hubUtils.retrieveModuleLibrary(moduleUri);
     } catch (e) {
-      httpUtils.throwBadRequest(`Unable to access module: ${moduleUri}. Verify that this module is in your modules database and that your user account has a role that grants read permission to this module.`);
+      if(e.stack && e.stack.includes("XDMP-MODNOTFOUND")){
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/step.sjs:119]: That looks good to me.

**Review by rjrudin — DISMISSED** [review]

**Comment by ryanjdew** [comment]
@srinathgit It looks like the expected error messages for a couple of unit tests elsewhere have failed. 

com.marklogic.hub.flow.impl.FlowRunnerTest#customStepReferencesModulePathThatDoesntExist
marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/impl/step/makeFunction.sjs

**Comment by srinathgit** [comment]
> @srinathgit It looks like the expected error messages for a couple of unit tests elsewhere have failed.
> 
> com.marklogic.hub.flow.impl.FlowRunnerTest#customStepReferencesModulePathThatDoesntExist
> marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/impl/step/makeFunction.sjs

Thanks Ryan. I think latest commit should fix them.

**Review by akshaysonvane — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Pull request merged**